lib/graphql/batch.rb

```diff
@@ -20,10 +20,15 @@ def self.batch
     end
 
     def self.use(schema_defn)
+      schema = schema_defn.target
       if GraphQL::VERSION >= "1.6.0"
-        schema_defn.instrument(:multiplex, GraphQL::Batch::SetupMultiplex)
+        instrumentation = GraphQL::Batch::SetupMultiplex.new(schema)
```

Comment:
What do you think of splitting the multiplex/query and field instrumenters in two different concerns? Setup does a lot right now I find.

Reply:
lib/graphql/batch/setup.rb is only 54 lines long, which seem quite related.  Also, when we want to break backwards compatibility, we can simplify the code by just having multiplex instrumentation and no need to share code between the two setup classes.

Reply:
makes sense!